src/modules/src/error_estimator_ukf.c

```diff
@@ -0,0 +1,1623 @@
+{
+  float q0_2 = quat[0] * quat[0];
+  float q1_2 = quat[1] * quat[1];
+  float q2_2 = quat[2] * quat[2];
+  float q3_2 = quat[3] * quat[3];
+
+  // compute elements of the direction cosine matrix, ned to body
+  dcm[0 * 3 + 0] = q0_2 + q1_2 - q2_2 - q3_2;
+  dcm[0 * 3 + 1] = 2 * (quat[1] * quat[2] + quat[0] * quat[3]);
+  dcm[0 * 3 + 2] = 2 * (quat[1] * quat[3] - quat[0] * quat[2]);
+
+  dcm[1 * 3 + 0] = 2 * (quat[1] * quat[2] - quat[0] * quat[3]);
+  dcm[1 * 3 + 1] = q0_2 - q1_2 + q2_2 - q3_2;
+  dcm[1 * 3 + 2] = 2 * (quat[2] * quat[3] + quat[0] * quat[1]);
+
+  dcm[2 * 3 + 0] = 2 * (quat[1] * quat[3] + quat[0] * quat[2]);
+  dcm[2 * 3 + 1] = 2 * (quat[2] * quat[3] - quat[0] * quat[1]);
+  dcm[2 * 3 + 2] = q0_2 - q1_2 - q2_2 + q3_2;
+}
+
+// transform quaternion into euler angles
+static void quatToEuler(float *quat, float *eulerAngles)
+{
+  float q0_2 = quat[0] * quat[0];
+  float q1_2 = quat[1] * quat[1];
+  float q2_2 = quat[2] * quat[2];
+  float q3_2 = quat[3] * quat[3];
+
+  // compute elements of the direction cosine matrix
+  float ele11 = q0_2 + q1_2 - q2_2 - q3_2;
+  float ele21 = 2 * (quat[1] * quat[2] + quat[0] * quat[3]);
+  float ele31 = 2 * (quat[1] * quat[3] - quat[0] * quat[2]);
+  float ele32 = 2 * (quat[2] * quat[3] + quat[0] * quat[1]);
+  float ele33 = q0_2 - q1_2 - q2_2 + q3_2;
+
+  //compute eulerAngles
+  if ((ele32 == 0) & (ele33 == 0))
+  {
+    eulerAngles[0] = 0;
+  }
+  else
+  {
+    eulerAngles[0] = atan2f(ele32, ele33);
+  }
+  eulerAngles[1] = asinf(-ele31);
+
+  if ((ele21 == 0) & (ele11 == 0))
+  {
+    eulerAngles[2] = 0;
+  }
+  else
+  {
+    eulerAngles[2] = atan2f(ele21, ele11);
+  }
+}
+
+// compute quaternion product
+static void multQuat(float *q1, float *q2, float *quatRes)
+{
+  quatRes[0] = q1[0] * q2[0] - q1[1] * q2[1] - q1[2] * q2[2] - q1[3] * q2[3];
+  quatRes[1] = q1[1] * q2[0] + q1[0] * q2[1] - q1[3] * q2[2] + q1[2] * q2[3];
+  quatRes[2] = q1[2] * q2[0] + q1[3] * q2[1] + q1[0] * q2[2] - q1[1] * q2[3];
+  quatRes[3] = q1[3] * q2[0] - q1[2] * q2[1] + q1[1] * q2[2] + q1[0] * q2[3];
+}
+
+// compute quaternion from attitude error
+static void quatFromAtt(float *attVec, float *quat)
+{
+  float absError = sqrtf(attVec[0] * attVec[0] + attVec[1] * attVec[1] + attVec[2] * attVec[2]);
+  float absError2 = absError * absError;
+  float absError3 = absError2 * absError;
+  float absError4 = absError3 * absError;
+
+  float scale = 0.5f - 0.02083333f * absError2 + 0.0002604166f * absError4;
+  quat[0] = 1.0f - 0.125f * absError2 + 0.002604166f * absError4;
+
+  quat[1] = scale * attVec[0];
+  quat[2] = scale * attVec[1];
+  quat[3] = scale * attVec[2];
+}
+
+// Temporary development groups
+LOG_GROUP_START(kalman_states)
+//  LOG_ADD(LOG_FLOAT, ox, &coreData.S[KC_STATE_X])
+//  LOG_ADD(LOG_FLOAT, oy, &coreData.S[KC_STATE_Y])
+//  LOG_ADD(LOG_FLOAT, vx, &coreData.S[KC_STATE_PX])
+//  LOG_ADD(LOG_FLOAT, vy, &coreData.S[KC_STATE_PY])
+LOG_GROUP_STOP(kalman_states)
+
+LOG_GROUP_START(navFilter)
```

Comment:
Kalman estimator here https://github.com/bitcraze/crazyflie-firmware/blob/af96c070d6bc32cf73eeb89980a970d7e01f7f8c/src/modules/src/estimator_kalman.c#L460-L469

This will look like [rendered like this](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#kalman) on the website:

Reply:
This is optional

Reply:
Hey, just to make sure to understand what you suggest:

You'd like to have logging names like it is done in the kalman-Filter Approach, e.g.
The State for the x-coordinate would read  ukf.stateX like kalman.stateX?

This is could definitely be renamed following this convention.